src/java/net/jpountz/lz4/LZ4BlockInputStream.java

```diff
@@ -68,6 +68,11 @@ public LZ4BlockInputStream(InputStream in, LZ4FastDecompressor decompressor, Che
     this.compressedBuffer = new byte[HEADER_LENGTH];
     o = originalLen = 0;
     finished = false;
+    stopOnEmptyBlock = true;
+  }
+
+  protected void setStopOnEmptyBlock(boolean stopOnEmptyBlock) {
```

Comment:
Should this be protected, or public?  I guess the use case you assume is to extend this class to enable the new behavior, but isn't it simpler to just make `setStopOnEmptyBlock()` public so that users can just call it if they want the new behavior?